src/components/datepicker/js/calendar.js

```diff
@@ -57,7 +57,8 @@
       scope: {
         minDate: '=mdMinDate',
         maxDate: '=mdMaxDate',
-        dateFilter: '=mdDateFilter'
+        dateFilter: '=mdDateFilter',
+        _currentView: '@mdCurrentView'
```

Comment:
Pretty weird. Not sure if you want to invest some more time, but you could possibly try to hook into that variable and try to figure out where it was changed from.